### PR TITLE
Arturotepaleaguilar/ch60/set up skills

### PR DESCRIPTION
**Pull request by atepag**
The only thing I could not mage to do was to automatically add the new added skill to both the project and/or the user so they would have to go back and add said skill from list. As for when posting a project a user can add a new skill to the list but the filled form resets so I didn't get around to that. Just added a small warning on top of the form explaining to check all the skills before filling the rest of the form.
--To add a user skill go to the profile page then click the Skills button.

### CODE REVIEW

File: ProjectNetwork.Services/ProjectsService.cs [hunk 1]
```diff
@@ -39,11 +53,28 @@ public Project GetProject(int Id)
             return project;
         }
 
+//<<<<<<< HEAD
+        public List<Skill> GetMySkills(int projectId)
+        {
+            var tempProject = this._projectSkillRepository.GetAll(proj => proj.ProjectId == projectId);
```
Comment (OleksandrVakhno) [inline ProjectNetwork.Services/ProjectsService.cs]: Why GetMySkills is part of the project service and not the skills service? And why are you getting "mySkills" from projectSkill repository and not userSkill?

File: ProjectNetwork.Services/SkillsService.cs [hunk 2]
```diff
@@ -12,13 +12,97 @@ public class SkillsService : ISkillsService
...
+            try
+            {
+               
+                if (skills != null)
+                {
+                    List<UserSkill> newSkills = new List<UserSkill>();
+                    foreach (int x in skills)
+                    {
+                        var cur = new UserSkill();
+                        cur.UserId = UserId;
```
Comment (OleksandrVakhno) [inline ProjectNetwork.Services/SkillsService.cs:56]: Check if the user with the specified ID exists.

File: ProjectNetwork.Services/SkillsService.cs [hunk 3]
```diff
@@ -12,13 +12,97 @@ public class SkillsService : ISkillsService
...
+            List<Skill> mySkills = new List<Skill>();
+            foreach (var userSkill in tempSkils)
+            {
+                mySkills.Add(GetASkill(userSkill.SkillId));
+            }
+            return mySkills;
+        }
+
+        
+        public bool PostSkills(string UserId, int[] skills)
```
Comment (OleksandrVakhno) [inline ProjectNetwork.Services/SkillsService.cs]: Find a better name. Why is it PostSkill and you are just assigning a skill to a user? Should it be something like "PostUserSkill"?

File: ProjectsNetwork/Areas/Identity/Pages/Account/Manage/ManageNavPages.cs [hunk 4]
```diff
@@ -40,6 +42,8 @@ public static class ManageNavPages
 
         public static string TwoFactorAuthenticationNavClass(ViewContext viewContext) => PageNavClass(viewContext, TwoFactorAuthentication);
 
+        //public static string SkillsNavClass(ViewContext viewContext) => PageNavClass(viewContext, Skills);
```
Comment (OleksandrVakhno) [inline ProjectsNetwork/Areas/Identity/Pages/Account/Manage/ManageNavPages.cs]: Delete unused code. No need to have a bunch of comments with the old code.

File: ProjectsNetwork/Areas/User/Controllers/SkillsController.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                return NotFound("User not found");
+            }
+            var mySkills = this._skillsService.GetMySkills(currentUserID);
+            var tupleModel = new Tuple<List<Skill>, List<Skill>, Skill>((List<Skill>)skills, mySkills, new Skill());
+            return View(tupleModel);
+        }
+
+        [HttpPost]
+        [Route("Post")]
+        public IActionResult Post(int[] skills)
```
Comment (OleksandrVakhno) [inline ProjectsNetwork/Areas/User/Controllers/SkillsController.cs]: Again, choose better names. I don't need to be going to code to read the difference between AddSkill and Post. It needs to be something like PostUserSkill and PostSkill

**Review by OleksandrVakhno — CHANGES_REQUESTED** [review]
Good. Minor issues. Left comments on what can be improved.

File: ProjectNetwork.Services/SkillsService.cs (hunk 2 above)
Comment (atepag) [reply ProjectNetwork.Services/SkillsService.cs:56]: SkillsController Line 50

**Comment by atepag** [comment]
Done!

**Review by OleksandrVakhno — APPROVED** [review]

**Pull request merged**